src/python/grpcio_observability/grpc_observability/server_call_tracer.h

```diff
@@ -0,0 +1,44 @@
+// Copyright 2023 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <grpc/support/port_platform.h>
+
+#include "absl/strings/string_view.h"
+#include "absl/strings/strip.h"
+
+#include "src/core/lib/slice/slice.h"
+#include "src/core/lib/channel/call_tracer.h"
+#include "src/core/lib/resource_quota/arena.h"
+
+#include "src/python/grpcio_observability/grpc_observability/python_census_context.h"
+#include "src/python/grpcio_observability/grpc_observability/observability_util.h"
+
+namespace grpc_observability {
+
+class PythonOpenCensusServerCallTracerFactory
+    : public grpc_core::ServerCallTracerFactory {
+ public:
+  grpc_core::ServerCallTracer* CreateNewServerCallTracer(
+      grpc_core::Arena* arena) override;
```

Comment:
They're not public API but he said we can use them in Python.

Reply:
If non-public Core APIs are a part of our API surface for the `ObservabilityPlugin` interface, then we cannot allow _any_ alternative implementations for the `ObservabilityPlugin` interface besides our `grpc_observability` package. This should be display very prominently in the docstrings.

Since this API also seems like it might not work cross-DLL on Windows, I think it may be better to provide an API that does not require use of the Arena for memory allocation.

CC @yashykt

Reply:
I added comment to mention it was only mean to be used by gRPC team.

As for the DDL issue, we decided not to used the provided arena.